src/Stratis.SmartContracts.CLR.Tests/SmartContracts/SystemContractsDictionary.cs

```diff
@@ -0,0 +1,205 @@
+﻿using Stratis.SmartContracts;
+
+public struct WhiteListEntry
+{
+    public UInt256 CodeHash;
+    public Address LastAddress;
+    public string Name;
+}
+
+public class SystemContractsDictionary : SmartContract
+{
+    const string primaryGroup = "main";
+
+    public SystemContractsDictionary(ISmartContractState state) : base(state)
+    {
+        this.State.SetArray($"Signatories:{primaryGroup}", new[] { new Address(0, 0, 0, 0, 0), new Address(0, 0, 0, 0, 1), new Address(0, 0, 0, 0, 2) });
+        this.State.SetUInt32($"Quorum:{primaryGroup}", 2);
+    }
+
+    public Address[] Signatories => GetSignatories(primaryGroup);
+
+    public uint Quorum => GetQuorum(primaryGroup);
+
+    private void VerifySignatures(string authorizationChallenge)
+    {
+        /*
+        Assert(this.VerifySignatures(System.Text.Encoding.ASCII.GetBytes(authorizationChallenge), signatures, this.Signatories).Length >= this.Quorum,
+            $"Please provide {this.Quorum} valid signatures for '{authorizationChallenge}'.");
+        */
+    }
+
+    public Address[] GetSignatories(string group)
+    {
+        Assert(!string.IsNullOrEmpty(group));
+        return this.State.GetArray<Address>($"Signatories:{group}");
+    }
+
+    public uint GetQuorum(string group)
+    {
+        Assert(!string.IsNullOrEmpty(group));
+        return this.State.GetUInt32($"Quorum:{group}");
+    }
+
+    public void AddSignatory(string group, Address address, uint newSize, uint newQuorum)
+    {
+        Assert(!string.IsNullOrEmpty(group));
+        Assert(newSize >= newQuorum, "The number of signatories can't be less than the quorum.");
+
+        Address[] signatories = this.GetSignatories(group);
+        for (int i = 0; i < signatories.Length; i++)
+            Assert(signatories[i] != address, "The signatory already exists.");
+
+        Assert((signatories.Length + 1) == newSize, "The expected size is incorrect.");
```

Comment:
Equality check should not  be negated in here?

Reply:
It is correct afaik.